Automattic/woocommerce-payments
# Migrate capabilities from test-drive to live account

Fixes 6924-gh-Automattic/transact-platform-server 

#### Changes proposed in this Pull Request

When a merchant transitions from a test-drive account to a live account, the capabilities for the payment methods enabled in test-drive mode will also be requested for the live account. 
This is done  as part of the onboarding process.
The purpose is to ensure a smooth configuration experience.









#### Testing instructions





* Spin up a new JN site from this PR's branch ( here is a [direct link](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/6924-migrate-test-drive-capabilities) to create it )
* checkout `add/6924-migrate-test-drive-capabilities` on platform, on local, run ngrok, and redirect the WCPay API requests to your local platform.
* Go to `Payments` from main menu, and under `I'm setting up a store for someone else.` section click the enable sandbox mode button . This will onboard a test-drive account.

<img width="1443" alt="Screenshot 2024-12-12 at 11 58 11" src="https://github.com/user-attachments/assets/c68b93a9-266d-4b6f-868f-b4f83827aa04" />

* After onboarding go to `Payments->Settings`, select other payment methods under `Payment methods` section, and click the save changes button

<img width="1119" alt="Screenshot 2024-12-12 at 12 04 30" src="https://github.com/user-attachments/assets/bd1d9d76-fa57-4ef8-8946-75c738e55ae3" />

* now go to `Payments -> Overview` and click on `set up a live Woopayments account` and complete the onboarding wizzard

<img width="1568" alt="Screenshot 2024-12-12 at 12 08 18" src="https://github.com/user-attachments/assets/6e94dcf5-fe75-4253-bd03-1a1ade5795b9" />

* after you finish the onboarding process, go to `Woocommerce -> Payments -> Settings` . The payment methods capabilities for the enabled payment methods in test-drive should be requested for the live account.

<img width="860" alt="Screenshot 2024-12-12 at 12 13 25" src="https://github.com/user-attachments/assets/70200cbb-2d08-47d6-b99d-e52c7106f7e0" />

* you can see the capabilities status also in WcPay dev tools. 

<img width="897" alt="Screenshot 2024-12-12 at 12 15 58" src="https://github.com/user-attachments/assets/e0ca5863-cdd7-44e9-8dc3-5cd9aabeb10d" />

* When the capabilities are approved by Stripe, there will be already enabled on client side. 


<img width="748" alt="Screenshot 2024-12-13 at 03 04 14" src="https://github.com/user-attachments/assets/f5d1416d-0974-495e-bb9f-ea9f33e8f618" />

-------------------


**Post merge**
